tools/notification-configuration/notification-creator/NotificationConfigurator.cs

```diff
@@ -7,29 +7,38 @@
 using Azure.Sdk.Tools.NotificationConfiguration.Enums;
 using Azure.Sdk.Tools.NotificationConfiguration.Models;
 using Azure.Sdk.Tools.NotificationConfiguration.Services;
-using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Azure.Sdk.Tools.NotificationConfiguration.Helpers;
+using System;
+using Azure.Sdk.Tools.CodeOwnersParser;
 
 namespace Azure.Sdk.Tools.NotificationConfiguration
 {
     class NotificationConfigurator
     {
         private readonly AzureDevOpsService service;
+        private readonly GitHubService gitHubService;
         private readonly ILogger<NotificationConfigurator> logger;
 
-        private const int MaxTeamNameLength = 64;
+        private const int MaxTeamNameLength = 64;        
+        // Type 2 maps to a pipeline YAML file in the repository
+        private const int PipelineYamlProcessType = 2;
+        // A cache on the code owners github identity to ms alias.
+        private readonly Dictionary<string, string> codeOwnerCache = new Dictionary<string, string>();
```

Comment:
This needs to be ConcurrentDictionary given your usage of the cache in async tasks.

Reply:
Done.